lib/matplotlib/backends/gtk3_compat.py

```diff
@@ -0,0 +1,24 @@
+try:
+    import gi
+except ImportError:
+    try:
+        import pgi as gi
+    except ImportError:
+        raise ImportError("Gtk3 backend requires pygobject to be installed.")
+
+try:
+    gi.require_version("Gtk", "3.0")
+except AttributeError:
+    raise ImportError(
+        "pygobject version too old -- it must have require_version")
+except ValueError:
+    raise ImportError(
+        "Gtk3 backend requires the GObject introspection bindings for Gtk 3 "
+        "to be installed.")
+
+try:
+    pkgs = ['Gtk', 'Gdk', 'GObject', 'GLib']
+    _temp = __import__(gi.__name__+'.repository', globals(), locals(), pkgs, 0)
+    Gtk, Gdk, GObject, GLib = _temp.Gtk, _temp.Gdk, _temp.GObject, _temp.GLib
+except (ImportError, AttributeError):
+    raise ImportError("Gtk3 backend requires pygobject to be installed.")
```

Comment:
I say cover it in this one, personally.  Might as well fix all of the Gtk3 import weirdness at once.